.github/workflows/scripts/build.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3 -u
```

Comment:
nit: hmmm this implicit deps on /usr/bin/python3 is a bit concerning, IMHO I'd prefer to be explicit about which python we use in CI :P

Reply:
Python running this build script and python actually running `setup.py` is not the same one. I'm writing this script with portability in mind, even dependencies are installed automatically, so don't worry :)

Reply:
Hmmm correct me if I'm wrong, when you run `.github/workflows/scripts/build.py` in testing.yml it's implicitly using `/usr/bin/python3` right? IMHO others who modify this script may wonder which python is being used?

Reply:
> Hmmm correct me if I'm wrong, when you run `.github/workflows/scripts/build.py` in testing.yml it's implicitly using `/usr/bin/python3` right? 

Right

> IMHO others who modify this script may wonder which python is being used?

There's no confusion here I think, `/usr/bin/python3` will point to system/container provided Python.